yarn-project/aztec-node/src/aztec-node/server.ts

```diff
@@ -704,10 +704,11 @@ export class AztecNodeService implements AztecNode {
    *
    * @param contract - Address of the contract to query.
    * @param slot - Slot to query.
+   * @param blockNumber - The block number at which to get the data or 'latest'.
    * @returns Storage value at the given contract slot.
    */
-  public async getPublicStorageAt(contract: AztecAddress, slot: Fr): Promise<Fr> {
-    const committedDb = await this.#getWorldState('latest');
+  public async getPublicStorageAt(contract: AztecAddress, slot: Fr, blockNumber: L2BlockNumber): Promise<Fr> {
```

Comment:
If you wanted to follow the same flow as before, why not make the default value of `blockNumber = 'latest'`?

Reply:
I'm not a fan of default values for these things tbh, specially when they'll later be exposed e.g. behind an http interface.